src/services/postServices.ts

```diff
@@ -0,0 +1,92 @@
+import { Document } from "mongoose";
+import { BadRequestError, ServerError } from "../errors/customErrors.ts";
+import PostModel, { Post } from "../models/Post.ts";
+import UserModel from "../models/UserModel.ts";
+
+export const getSinglePostDocument = async (
+  postId: string,
+): Promise<Document> => {
+  try {
+    const post = await PostModel.findById(postId);
+    if (!post) {
+      throw new ServerError("Post not found");
+    }
+    return post.toJSON() as Document;
+  } catch (err) {
+    throw new ServerError(`${err}`);
+  }
+};
+
+export const buildPostDocument = async (body: Post): Promise<string> => {
+  try {
+    const { id, author } = body;
+    const getUser = await UserModel.findById(author);
+
+    if (id) {
+      const existedPost = await PostModel.findById(id);
+      if (existedPost) {
+        throw new BadRequestError(`Post already exists`);
+      }
+    }
+    if (!getUser) {
+      console.log("here");
+      throw new BadRequestError("Invalid author");
+    }
+    if (!body.body || !body.header) {
+      throw new BadRequestError("Missing information");
+    }
+    const newPost = await PostModel.create({
+      author: body.author,
+      header: body.header,
+      body: body.body,
+      media: body.media,
+      privacy: body.privacy,
+    });
+
+    return newPost._id as string;
+  } catch (error) {
+    throw new ServerError(`${error}`);
+  }
+};
+export const updatePostDocument = async (
+  postId: string,
+  body: Post,
+): Promise<string> => {
+  try {
+    const updatePost = await PostModel.findById(postId);
```

Comment:
This works but you could refer the following update method for cleaner 
    const { title, description, area, location, company, link, salary, geometry, state,generate } = req.body;
    const internshipData = { title: title, description: description, area: area, location: location, state: state, company: company, link: link, salary: salary, geometry: geometry }
    const internship = await Internship.findByIdAndUpdate(id, { ...internshipData });
    One line edit.

Reply:
> This works but you could refer the following update method for cleaner const { title, description, area, location, company, link, salary, geometry, state,generate } = req.body; const internshipData = { title: title, description: description, area: area, location: location, state: state, company: company, link: link, salary: salary, geometry: geometry } const internship = await Internship.findByIdAndUpdate(id, { ...internshipData }); One line edit.

Hi Hung I updated the method at [#d42436d](https://github.com/hungngodev/Flavorie/pull/82#discussion_r1634133914)